flask_classful.py

```diff
@@ -228,6 +236,12 @@ def proxy(**forgettable_view_args):
                         response = cls.representations[list(cls.representations.keys())[0]](
                             response, code, headers)
 
+            # If the header or code is set, regenerate the response
+            elif any(x is not sentinel for x in (code, headers)):
+                # A response can be passed into `make_response` and it will set
+                # the key appropriately
+                response = make_response(response, code, headers)
```

Comment:
Also, we should add a test that doesn't return a `tuple`, and just returns a Flask Response, because that would have totally caught this bug.